### ISSUE TITLE
GUI Window Re-scaling Bug

### ISSUE THREAD

**Issue by Indoorjoker** [ModOrganizer2/modorganizer#895]
## The problem:
Mod Organizer 2 GUI can sometimes shift off the screen when re-scaling the window. This affects sub-menus and is constant after restarting the application.

## To Reproduce:
Steps to reproduce: Whilst the menu is not full-screen attempt to resize the window from the top of the menu. 

## Environment:

* Mod Organizer Version that exhibits the issue: V2.2.1
* Desktop OS/version used to run Mod Organizer: Windows 10

## Details:
It is possible to undo this by closing the program and changing your display resolution in Windows Settings then starting the application again.

This is an mp4 video of what happens when you re-scale windows. As shown the interface seems to fall off the screen never to return even on closing and restart, this can affect sub-menus and not just the main interface.

[Bug.zip](https://github.com/ModOrganizer2/modorganizer/files/3881554/Bug.zip)

**Comment by isanae** [comment]
I've been chasing bugs like this for a while, but I can't reproduce what you're getting. There should be a fix in the next version that will put the window back on screen after a restart. In the meantime, can you upload your `mo_interface.log` here? I'd also need some information about your setup: how many monitors, any high dpi stuff or varied resolutions, etc.

**Comment by Indoorjoker** [comment]
> I've been chasing bugs like this for a while, but I can't reproduce what you're getting. There should be a fix in the next version that will put the window back on screen after a restart. In the meantime, can you upload your `mo_interface.log` here? I'd also need some information about your setup: how many monitors, any high dpi stuff or varied resolutions, etc.

Hey thanks for the quick reply, my setup is only only one 1080p monitor set at its native resolution 1920 x 1080. In reality this is just an old flat-screen so I doubt there's any high DPI display shenanigans. Display scaling is set to 150% in windows settings as well.

**Comment by Al12rs** [comment]
Don't do it if you didn't already, in case isa has more tests to ask you to reproduce, but did you reset your window geometries from the Mo2 settings after the last Mo2 update? That usually fixes these kind of issues but if we can avoid them all together that would be better.

**Comment by Indoorjoker** [comment]
> Don't do it if you didn't already, in case isa has more tests to ask you to reproduce, but did you reset your window geometries from the Mo2 settings after the last Mo2 update? That usually fixes these kind of issues but if we can avoid them all together that would be better.

This is actually a fresh install from the nexus page by LePresidente, I never updated Mo2 so I believe this is unique at least in my case. I only installed and ran into the bug today(11/22/2019), unfortunately.
 
Link Here: https://www.nexusmods.com/skyrimspecialedition/mods/6194?tab=description

**Comment by Al12rs** [comment]
Thanks for the answer, it might be something to try later to see whether that fixes it anyways. I'll leave it to isa if he needs anything else.

**Comment by Indoorjoker** [comment]
For sure I'll keep a lookout if there's anything that's needed from me, any reply's will likely update me in 15 to 20 minutes afterwards through email.

**Comment by isanae** [comment]
> Display scaling is set to 150% in windows settings as well

Holy cow, that's an instant reproduce. I'll be having a look at this shortly. Thanks!

**Comment by Indoorjoker** [comment]
Yeah 32 inch display needs a lot of scaling to make things near readable on my end. For me it looks normal but can have some issues with small or normal text that isn't scaled.

**Comment by Indoorjoker** [comment]
Ah, so you reproduced the problem? So should I close the issue or keep it open? If there's anything you need from me just shoot me a comment.

**Comment by isanae** [comment]
Keep it open until it's fixed. I have a feeling it's a bug in Qt itself though, so there might not be much I can do. We'll see.

**Comment by Indoorjoker** [comment]
> Keep it open until it's fixed. I have a feeling it's a bug in Qt itself though, so there might not be much I can do. We'll see.

Alright, will do.

**Comment by Silarn** [comment]
To be clear, Qt's native auto scaling has always been a bit problematic. I did create a branch a while back to utilize dynamic scaling by adjusting the default sizes based on the scale but it was very much a WIP. In particular we would need high res variations of all of our icons for high dpi displays and so forth.

But that's overall a better solution to scaling than the built-in version.

**Comment by selsie** [comment]
I can't reproduce the problem. The resizing of all the menus stutters, but that's not a big deal.
